prover/src/lib.rs

```diff
@@ -283,23 +276,22 @@ pub trait Prover {
         let mut aux_trace_segments = Vec::new();
         let mut aux_trace_rand_elements = AuxTraceRandElements::new();
         for i in 0..trace.layout().num_aux_segments() {
-            #[cfg(feature = "std")]
-            let now = Instant::now();
-
-            // draw a set of random elements required to build an auxiliary trace segment
-            let rand_elements = channel.get_aux_trace_segment_rand_elements(i);
-
-            // build the trace segment
-            let aux_segment = trace
-                .build_aux_segment(&aux_trace_segments, &rand_elements)
-                .expect("failed build auxiliary trace segment");
-            #[cfg(feature = "std")]
-            debug!(
-                "Built auxiliary trace segment of {} columns and 2^{} steps in {} ms",
-                aux_segment.num_cols(),
-                aux_segment.num_rows().ilog2(),
-                now.elapsed().as_millis()
-            );
+            let num_columns = trace.layout().get_aux_segment_width(i);
+            let (aux_segment, rand_elements) = {
+                let _ = info_span!("build_aux_trace_segment", num_columns).entered();
+
+                // draw a set of random elements required to build an auxiliary trace segment
+                let rand_elements = channel.get_aux_trace_segment_rand_elements(i);
+
+                // build the trace segment
+                let aux_segment = trace
+                    .build_aux_segment(&aux_trace_segments, &rand_elements)
+                    .expect("failed build auxiliary trace segment");
+
+                (aux_segment, rand_elements)
+            };
+            assert_eq!(aux_segment.num_cols(), num_columns);
+            assert_eq!(aux_segment.num_rows(), air.trace_length());
 
             // extend the auxiliary trace segment and build a Merkle tree from the extended trace
             let (aux_segment_polys, aux_segment_root) =
```

Comment:
For the code below, let's do the same thing as for the previous comment. The high-level span can be called `commit_to_aux_trace_segment`.